app/code/Magento/ConfigurableProductGraphQl/Model/Resolver/Variant/Attributes.php

```diff
@@ -35,12 +45,12 @@ public function resolve(
         $data = [];
         foreach ($value['options'] as $option) {
             $code = $option['attribute_code'];
-            if (!isset($value['product'][$code])) {
+            if (!isset($value['product']['model'][$code])) {
```

Comment:
Looks like we have a problem at this place because `$value['product']['model']` contains Product model object